README.md

```diff
@@ -0,0 +1,257 @@
+# Brigade Docker Hub Gateway
+
+This is a work-in-progress
+[Brigade 2](https://github.com/brigadecore/brigade/tree/v2)
+compatible gateway that receives events (webhooks) from Docker Hub
+and propagates them into Brigade 2's event bus.
+
+## Installation
+
+Prerequisites:
+
+* A Kubernetes cluster:
+    * For which you have the `admin` cluster role
+    * That is already running Brigade 2
+    * Capable of provisioning a _public IP address_ for a service of type
+      `LoadBalancer`. (This means you won't have much luck running the gateway
+      locally in the likes of kind or minikube unless you're able and willing to
+      mess with port forwarding settings on your router, which we won't be
+      covering here.)
+
+* `kubectl`, `helm` (commands below assume Helm 3), and `brig` (the Brigade 2
+  CLI)
+
+### 1. Create a Service Account for the Gateway
+
+__Note:__ To proceed beyond this point, you'll need to be logged into Brigade 2
+as the "root" user (not recommended) or (preferably) as a user with the `ADMIN`
+role. Further discussion of this is beyond the scope of this documentation.
+Please refer to Brigade's own documentation.
+
+Using Brigade 2's `brig` CLI, create a service account for the gateway to use:
+
+```console
+$ brig service-account create \
+    --id brigade-dockerhub-gateway \
+    --description brigade-dockerhub-gateway
+```
+
+Make note of the __token__ returned. This value will be used in another step.
+_It is your only opportunity to access this value, as Brigade does not save it._
+
+Authorize this service account to create new events:
+
+```console
+$ brig role grant EVENT_CREATOR \
+    --service-account brigade-dockerhub-gateway \
+    --source brigade.sh/dockerhub
+```
+
+__Note:__ The `--source brigade.sh/dockerhub` option specifies that
+this service account can be used _only_ to create events having a value of
+`brigade.sh/dockerhub` in the event's `source` field. _This is a
+security measure that prevents the gateway from using this token for
+impersonating other gateways._
+
+### 2. Install the Docker Hub Gateway
+
+For now, we're using the [GitHub Container Registry](https://ghcr.io) (which is
+an [OCI registry](https://helm.sh/docs/topics/registries/)) to host our Helm
+chart. Helm 3 has _experimental_ support for OCI registries. In the event that
+the Helm 3 dependency proves troublesome for users, or in the event that this
+experimental feature goes away, or isn't working like we'd hope, we will revisit
+this choice before going GA.
+
+To fetch the Brigade Docker Hub Gateway chart from the registry:
+
+```console
+  export HELM_EXPERIMENTAL_OCI=1
+  helm chart pull ghcr.io/brigadecore/brigade-dockerhub-gateway:v2.0.0-alpha.1
```

Comment:
As this is a brand new gateway (not a rewrite of an existing one), do we still want to start versioning w/ `v2.0.0-*`?  Naturally, this versioning does align with Brigade v2.  I could also see it starting with the usual `v0.1.0-*`.  No real preference on my end, just wanted to check.

Reply:
It should have been v0.1.0 initially.

Reply:
I'll fix.